main.rb

```diff
@@ -0,0 +1,119 @@
+# rubocop: disable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
+module Enumerable
+  def my_each
+    return enum_for(:my_each) unless block_given?
+
+    i = 0
+    while i < length
+      yield(self[i])
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return enum_for(:my_each_with_index) unless block_given?
+
+    i = 0
+    while i < length
+      yield(self[i], i)
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return enum_for(:my_select) unless block_given?
+
+    items_selected = []
+    my_each { |item| items_selected << item if yield(item) }
+    items_selected = items_selected.to_h if is_a?(Hash)
+    items_selected
+  end
+
+  def my_all?(*args)
+    if !args[0].nil?
+      my_each { |item| return false unless item.is_a?(args[0]) }
+    elsif block_given?
+      my_each { |item| return false unless yield(item) }
+    else
+      my_each { |item| return false unless item }
+    end
+    true
+  end
+
+  def my_any?(*args)
+    if !args[0].nil?
+      my_each { |item| return true if item.is_a?(args[0]) }
+    elsif block_given?
+      my_each { |item| return true if yield(item) }
+    else
+      my_each { |item| return true if item }
+    end
+    false
+  end
```

Comment:
- [ ] in my_any? when a Regex is passed as an argument returns false if none of the collection matches the Regex

%w[dog door rod blade].my_any?(/z/) should be equal to %w[dog door rod blade].any?(/z/) 

- [ ] also, when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern

`%w[dog door rod blade].my_any?('cat') `should be equal to `%w[dog door rod blade].any?('cat') `